lib/core/errors.js

```diff
@@ -88,13 +88,23 @@ class InformationalError extends UndiciError {
   }
 }
 
-class ContentLengthMismatchError extends UndiciError {
+class RequestContentLength extends UndiciError {
```

Comment:
All other errors end with `Error`. Also doesn't feel/sound like a descriptive error like this.

Reply:
I think it's worth having them apart, but I'm ok to merge them